### DISCUSSION TITLE
ALERT: ALERT_PEER_AUTH_FAILED!

### DISCUSSION THREAD

**Discussion by rootBiao** [strongswan/strongswan#1803] [Q&A]
My  strongswan server's log has encountered this error message. What is the likely cause?

Jul 19 15:15:39 16[IKE] <casa|16> Created EAP 5G Process Server.
Jul 19 15:15:39 09[CFG] <casa|16> ALERT: ALERT_PEER_AUTH_FAILED!
Jul 19 15:15:39 09[CFG] <casa|16> AUTH Failure.
Jul 19 15:15:39 09[IKE] <casa|16> WARNING: IKE Down notification received in DELETE_STATE_NULL. Send IKE_DOWN to tunnelizer. initiator SPI 6ac67058783d1f23 Message ID's I:0 R:5.
Jul 19 15:15:39 09[CFG] <casa|16> IKE 10 SK_EI: => 16 bytes @ 0x7fcfa8004710
Jul 19 15:15:39 09[CFG] <casa|16>    0: 20 C9 D0 84 57 0B 6D B3 01 72 18 0F 0A CC 02 2D   ...W.m..r.....-
Jul 19 15:15:39 09[CFG] <casa|16> IKE 10 SK_ER: => 16 bytes @ 0x7fcfa8007dd0
Jul 19 15:15:39 09[CFG] <casa|16>    0: 1F 29 8A 09 00 0D 10 94 45 47 5A 33 E3 CC F0 78  .)......EGZ3...x
Jul 19 15:15:39 09[CFG] <casa|16> IKE 10 SK_AI: => 20 bytes @ 0x7fcfa8007a60
Jul 19 15:15:39 09[CFG] <casa|16>    0: DD FA D5 AE C6 76 0F C8 AA CD D6 CC 5C EB 88 11  .....v......\...
Jul 19 15:15:39 09[CFG] <casa|16>   16: CC 4E FD 2B                                      .N.+
Jul 19 15:15:39 09[CFG] <casa|16> IKE 10 SK_AR: => 20 bytes @ 0x7fcfa8007ae0
Jul 19 15:15:39 09[CFG] <casa|16>    0: 02 BC 44 CD 9B D7 A6 DC 67 81 69 BB C4 2E 12 C7  ..D.....g.i.....
Jul 19 15:15:39 09[CFG] <casa|16>   16: F7 57 66 40

**Comment by tobiasbrunner** [comment]
A client with invalid credentials? Read the log before that alert was triggered.

**Reply by rootBiao** [reply]
![image](https://github.com/strongswan/strongswan/assets/87700793/dc5943aa-e46e-4892-87f2-3bef35f44b04)
My strongswan version is tag: 5.8.4.
Does the client require credentials?

**Reply by tobiasbrunner** [reply]
What do you mean? It obviously needs some kind of credentials (certificate/private key, username/password, PSK etc. depending on the configured authentication method).

**Reply by rootBiao** [reply]
**This  is the configuration of my client strongswan. conf**

charon {
	load_modular = yes
	threads=48
	keep_alive = 600
	    reuse_ikesa = no

     filelog {
        charon {
            path = /var/log/charon.log
            time_format = %b %e %T
            ike_name = yes
            append = no
            default = 4
            flush_line = yes
          }
        }

	plugins {

	    eap5g {
              simgnb_mode=yes
              unique_id_modifier=0
            }


	    load-tester {
            	# enable the plugin
            	enable = yes
            	# 10000 connections, ten in parallel
            	initiators = 1
            	iterations = 1
		delay = 10
                addrs {
                #     # eth0 = 20.0.0.103/16
                     eno3 = 10.0.0.103/16
	
					

                }
	    #responder = 20.0.0.10
		responder =  20.0.0.1
		initiator_tsi = 0.0.0.0/0
		initiator_tsr = 0.0.0.0/0
		proposal = aes128-sha1-modp1024
                esp = aes128-sha1-modp1024
		initiator_auth = eap5g
                initiator_id=1000000000%d@strongswan.org
            	responder_auth = psk
		eap_password = psk1
            	preshared_key = casa
            # request a virtual IP using configuration payloads
            	request_virtual_ip = yes
            	install_virtual_ip = no
           # disable IKE_SA rekeying (default)
            	ike_rekey = 2592000
            # enable CHILD_SA every 60s
            	child_rekey = 2592000
            # do not delete the IKE_SA after it has been established (default)
            	delete_after_established = no
            # do not shut down the daemon if all IKE_SAs established
            	shutdown_when_complete = no
            	dpd_delay = 2592000
	    }
	  include strongswan.d/charon/*.conf

	}
}
include strongswan.d/*.conf


=================================================================================
**This is the configuration of my server ipsec.secrets**
: PSK casa
=====================================


**This  is the configuration of my server strongswan. conf**

charon {
        load_modular = yes
        threads = 16
        block_threshold = 10000
        keep_alive = 600
	ikesa_table_size = 4096
    	ikesa_table_segments = 16
	dos_protection=no
	make_before_break = yes

     filelog {
        charon {
            path = /var/log/charon.log
            time_format = %b %e %T
            ike_name = yes
            append = yes
            default = 4
            flush_line = yes
          }
        }

	plugins {
		include strongswan.d/charon/*.conf

		ha {
      		   local = 0
      		   remote = 10.0.0.1
		   segment_count = 2
      		   # secret = s!ronG-P5K-s3cret
		   fifo_interface = yes
      		   monitor = yes
      		   resync = yes
    		}

	        eap5g {
                  simgnb_mode=no
		  zero_msk=no
	          eap5g_port="38123"
	          nas_tcp_port="38413"
	        }
                vici {
                        socket="tcp://127.0.0.1:4502"
                }
    }
}

include strongswan.d/*.conf



============================================================
**This is the configuration of my server ipsec.secrets**
: PSK casa

**Reply by rootBiao** [reply]
But I can't find which configuration is incorrect.

**Reply by tobiasbrunner** [reply]
I can only repeat myself, read the logs! These are the global config files, not the connection configs (which are stored in swanctl.conf). But I also see a lot of special stuff in these files (custom plugin, load-tester config, ha plugin config).

**Reply by rootBiao** [reply]
![image](https://github.com/strongswan/strongswan/assets/87700793/69824f70-92fe-4aa4-8f01-d2464a97b76f)
Excuse me, is the key configuration incorrect?
